src/core/PancakeInfinity.sol

```diff
@@ -511,24 +526,40 @@ abstract contract PancakeInfinity is SettlerAbstract {
                 if (uint256(poolManagerId) == 0) {
                     poolKey.poolManager = CL_MANAGER;
 
-                    delta = IPancakeInfinityCLPoolManager(address(poolKey.poolManager)).unsafeSwap(
+                    uint256 priceSqrtX96 = CL_MANAGER.unsafeSqrtPriceX96(poolKey);
+                    {
+                        // uint256 used in favor of future operations
+                        // value is uint160
+                        // Factor is:
+                        // 28011385487393067476124172288 approximately 1 / sqrt(2) in Q65.95 (95 bits)
+                        // 56022770974786143748341366784 approximately sqrt(2) in Q65.95 (96 bits)
+                        // Q65.95 was used instead of Q64.96 to prevent uint256 overflows later on as sqrt(2) in Q64.96 would be 97 bits
+                        uint256 factor = zeroForOne.ternary(uint256(28011385487393067476124172288), uint256(56022770974786143748341366784));
+
+                        unchecked {
+                            // no overflow when multiplying as factors are 160 bits and at most 96 bits respectively
+                            // shifted right 95 bitsto keep the price as Q64.96
+                            priceSqrtX96 = (priceSqrtX96 * factor) >> 95;
+                        }
+                        
+                        uint256 limit = zeroForOne.ternary(uint256(4295128740), uint256(1461446703485210103287273052203988822378723970341));
+                        (uint256 lo, uint256 hi) = zeroForOne.maybeSwap(priceSqrtX96, limit);
+                        priceSqrtX96 = uint160((lo > hi).ternary(limit, priceSqrtX96));
+                    }
+
+                    delta = CL_MANAGER.unsafeSwap(
                         poolKey,
                         zeroForOne,
                         amountSpecified,
-                        // TODO: price limits
-                        uint160(
-                            zeroForOne.ternary(
-                                uint160(4295128740), uint160(1461446703485210103287273052203988822378723970341)
-                            )
-                        ),
+                        priceSqrtX96,
                         hookData
                     );
                 } else if (uint256(poolManagerId) == 1) {
                     poolKey.poolManager = BIN_MANAGER;
                     if (amountSpecified >> 127 != amountSpecified >> 128) {
                         Panic.panic(Panic.ARITHMETIC_OVERFLOW);
                     }
-                    delta = IPancakeInfinityBinPoolManager(address(poolKey.poolManager)).unsafeSwap(
+                    delta = BIN_MANAGER.unsafeSwap(
```

Comment:
What is the purpose of this change? AFAIK these are equivalent, but the former expression significantly improves on contract size.